.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,15 @@
+name: Linting and formatting (pre-commit)
+
+on:
+  push:
+    branches: [ main ]
```

Comment:
Technically, this means that the operations will be run only when pushing to the main branch, so in a PR?
Or will a PR be rejected if it does not conform to the syntax?

Reply:
It will run on a PR. Currently, it will say "failed" but it's still possible to merge.